awslabs/aws-sdk-swift
# feat: Bearer token auth

Companion PR: https://github.com/smithy-lang/smithy-swift/pull/786

## Issue \#

https://github.com/awslabs/aws-sdk-swift/issues/1083

## Description of changes

Refer to comments on github diff

## Manual testing steps
To test this locally, first set up your IAM Identity Center ([doc link](https://docs.aws.amazon.com/singlesignon/latest/userguide/get-set-up-for-idc.html)) & AWS CodeCatalyst ([doc link](https://docs.aws.amazon.com/codecatalyst/latest/userguide/setting-up-topnode.html)).
Then, configure your config file to contain the following:
```
[default]
region=us-west-2
output=json
sso_session = codecatalyst

[profile codecatalyst]
region = us-west-2
sso_session = codecatalyst

[sso-session codecatalyst]
sso_region = us-east-1
sso_start_url = https://view.awsapps.com/start
sso_registration_scopes = codecatalyst:read_write
```
Then, run 
```
aws sso login --profile=codecatalyst
```
on the terminal. This will open a browser tab for you to authorize your AWS account to be accessed by CodeCatalyst.
Successful previous step creates a token file at path `~/.aws/sso/cache/<sha-1-hash-of-sso-sesion-name>.json`.
In CLITool, use `CodeCatalyst::VerifySession` API by simply doing the following:
```
func run() async throws {
        let client = try CodeCatalystClient(region: "us-west-2")
        let id = try await client.verifySession(input: VerifySessionInput()).identity
        print(id ?? "#### NO_ID_RETRIEVED")
}
```
Observe that the server accepts bearer auth and returns the ID successfully.

## New/existing dependencies impact assessment, if applicable



## Conventional Commits



By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
